### PR TITLE
Feature: Add _spec suffix presence checker

### PR DESCRIPTION
**Pull request by basherru**

### CODE REVIEW

File: lib/ci_helper/commands/check_spec_suffixes.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        base_paths.select do |path|
+          next if path.start_with?("spec/support")
+          next if path.start_with?("spec/factories")
+          next if path.end_with?("context.rb")
+          true
+        end
+      end
+
+      def base_paths
+        Dir["spec/*/**/*.rb"]
```
Comment (AnotherRegularDude) [inline lib/ci_helper/commands/check_spec_suffixes.rb:28]: Why do we ignore files in the root of spec folder?

if i write `spec/ci_helper.rb` (want `spec/ci_helper_spec.rb`), this command doesn't detect that _spec postfix was missed.

File: lib/ci_helper/version.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 
 module CIHelper
-  VERSION = "0.1.1"
+  VERSION = "0.2.1"
```
Comment (AnotherRegularDude) [inline lib/ci_helper/version.rb]: `0.2.0`?

File: lib/ci_helper/commands/check_spec_suffixes.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module CIHelper
+  module Commands
+    class CheckSpecSuffixes < BaseCommand
+      def call
+        target_paths.reject { |path| path.end_with?("_spec.rb") }.tap do |paths|
```
Comment (AnotherRegularDude) [inline lib/ci_helper/commands/check_spec_suffixes.rb]: Why do we use `#tap` method here?
We can't just assign to the var?

```ruby
paths = target_paths.reject { |path| path.end_with?("_spec.rb") }
```

File: README.md [hunk 4]
```diff
@@ -54,6 +54,13 @@ Accepted flags: `--node-index`, `node-total`, `with-database`, `split-resultset`
     * `--split-resultset` — if you run specs in parallel in CI,
         then you might use this flag to `true`. If this flag set to true,
         final `.resultset.json` will be renamed to `.resultset.#{node_index}.json`
+* **CheckSpecSuffixes** — checks specs in the spec subdirectories for `_spec` suffix,
+    by default ignored directories `support`, `factories` and files with `_context` suffix.
```
Comment (tycooon) [inline README.md]: ignores?

File: lib/ci_helper/commands/check_spec_suffixes.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+      def target_paths
+        spec_paths + extra_paths - ignored_paths
+      end
+
+      def spec_paths
+        base_paths.select do |path|
+          next if path.start_with?("spec/support")
+          next if path.start_with?("spec/factories")
+          next if path.end_with?("context.rb")
```
Comment (tycooon) [inline lib/ci_helper/commands/check_spec_suffixes.rb:22]: В ридми написано про суффикс `_context` 🤔

**Review by AnotherRegularDude — COMMENTED** [review]
Let's add this check to github actions of this repo?

File: lib/ci_helper/commands/check_spec_suffixes.rb (hunk 1 above)
Comment (AnotherRegularDude) [reply lib/ci_helper/commands/check_spec_suffixes.rb:29]: Or let's write in description, that files in `spec/*` is ignored by default

Comment (AnotherRegularDude) [reply lib/ci_helper/commands/check_spec_suffixes.rb:28]: Or let's write in description, that files in `spec/*` is ignored by default

File: lib/ci_helper/version.rb (hunk 2 above)
Comment (basherru) [reply lib/ci_helper/version.rb]: :pepega:

File: lib/ci_helper/commands/check_spec_suffixes.rb (hunk 1 above)
Comment (basherru) [reply lib/ci_helper/commands/check_spec_suffixes.rb:28]: It is a rare case when spec files are placed in the root of `spec` folder, usually there are subfolders.
Also in the spec folder root there are usually many helpers that should not be scanned.

In the doc it is stated that it `checks specs in the spec subdirectories`.

For the one willing to include the root of `spec` folder there is a `extra_paths` options.

File: lib/ci_helper/commands/check_spec_suffixes.rb (hunk 3 above)
Comment (basherru) [reply lib/ci_helper/commands/check_spec_suffixes.rb]: tap ftw

**Pull request merged**